src/main/java/com/yugabyte/servicebroker/utils/YSQLClient.java

```diff
@@ -0,0 +1,142 @@
+/* Copyright (c) YugaByte, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied.  See the License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.yugabyte.servicebroker.utils;
+
+import com.google.common.net.HostAndPort;
+import com.yugabyte.servicebroker.repository.YugaByteConfigRepository;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.yugabyte.servicebroker.utils.CommonUtils.generateRandomString;
+
+public class YSQLClient extends YBClient {
+  private static final Log logger = LogFactory.getLog(YSQLClient.class);
+
+  private static String DEFAULT_YSQL_USER = "yugabyte";
+  private static String DEFAULT_YSQL_PASSWORD = "yugabyte";
+  private String ADMIN_CREDENTIAL_KEY = "ysql-admin-user";
+  private static int DEFAULT_YSQL_PORT = 5433;
+
+  private enum YSQLRole  {
+    SUPERUSER("superuser_role"),
+    NOSUPERUSER("user_role");
+
+    public String value;
+    private YSQLRole(String role) {
+        this.value = role;
+    }
+  }
+
+  private Connection connection;
+
+  public YSQLClient(List<HostAndPort> serviceHosts,
+                    YugaByteConfigRepository yugaByteConfigRepository) {
+    super(serviceHosts, yugaByteConfigRepository);
+  }
+
+  private void getConnection() {
+    HostAndPort initialHostPort = getServiceHostPorts().get(0);
+    String jdbcUrl = String.format("jdbc:postgresql://%s:%d/%s", initialHostPort.getHostText(),
```

Comment:
why are we adding the `yugabyte` user, the last param is actually database right?